src/source/Common/IotCredentialProvider.c

```diff
@@ -176,10 +176,20 @@ STATUS parseIotResponse(PIotCredentialProvider pIotCredentialProvider, PCallInfo
         pIotCredentialProvider->pAwsCredentials = NULL;
     }
 
-    // Fix-up the expiration to be no more than max enforced token rotation to avoid extra token rotations
-    // as we are caching the returned value which is likely to be an hour but we are enforcing max
-    // rotation to be more frequent.
-    expiration = MIN(expiration, currentTime + MAX_ENFORCED_TOKEN_EXPIRATION_DURATION);
+    // Need to have large numbers since all items are in terms of HUNDREDS_NANOS_IN_A_SECOND. -1 to round down
+    randMultiplier = (ULONG_MAX / RAND_MAX) - 1;
+
+    // add randomized jitter between 1-20% of expiration
+    srand(currentTime / ((UINT64) sessionToken[rand() % (sessionTokenLen - 1)] + (UINT64) sessionToken[rand() % (sessionTokenLen - 1)]));
```

Comment:
Can (UINT64) sessionToken[rand() % (sessionTokenLen - 1)] be 0?

Reply:
No. It's a JSON string, not a binary data, being converting to a cstring, the -1 on the length prevents the null terminating character at the end from being included.

Reply:
No it's a JSON string, and JSON string's can't contain null terminating characters in the middle of the string. The last character is omitted with the -1.
https://docs.aws.amazon.com/STS/latest/APIReference/API_Credentials.html

API showing it's a JSON string returned.